src/Microsoft.AspNet.OData.Shared/Builder/ParameterConfiguration.cs

```diff
@@ -51,5 +51,44 @@ protected ParameterConfiguration(string name, IEdmTypeConfiguration parameterTyp
         /// Gets or sets a value indicating whether this parameter is nullable or not.
         /// </summary>
         public bool Nullable { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this parameter is optional or not.
+        /// </summary>
+        public bool Optional { get; set; }
+
+        /// <summary>
+        /// Gets or sets a default value for optional parameter.
+        /// </summary>
+        public string DefaultValue { get; set; }
+
+        /// <summary>
+        /// Sets the optional value as true.
+        /// </summary>
+        public ParameterConfiguration IsOptional()
```

Comment:
I think "SetOptional" would be more clear for the name because when I see "IsOptional", I expect it to be returning a boolean and not performing any updates.

I also think we should make the Optional.set private because we have this custom function to set the property value.

Reply:
Thanks. 
I think it's better to switch both.
the property name should be "IsOptional {get, private set;}, while the fluent API should be "Option()", "Required()".  I think that's Web API OData fluent API naming convention. You maybe find a lot of such fluent APIs defined in the model part.
I will change it.

Reply:
:+1:

Reply:
#Resolved.